### DISCUSSION TITLE
Assembly reference editing

### DISCUSSION THREAD

**Discussion by MadProbe** [dnSpyEx/dnSpy#158] [Q&A] [answered]
How can I edit assembly's references? I have searched everything and I couldn't find it.

**Comment by GazziFX** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/43498426/212390172-27fc4526-deeb-452a-8a46-147466a77ae1.png)

**Comment by MadProbe** [comment]
But also another question: How can I *delete* the reference (so I could remove class conflicts in method editor because one of referenced classes is resolved in 2 completely separate assemblies)

**Reply by ElektroKill** [reply]
You cannot just delete an assembly reference. There could've several TypeRef rows, or other rows which reference a resolution scope, pointing to it which would become invalid upon deletion. For more complex editing like resolving assembly reference conflicts, I suggest you write code using dnlib to perform this task. You should be able to use dnlib to get all TypeRefs and other objects which contain a resolution scope, make sure that it does not reference the AssemblyRef you are trying to remove, and then just save the module. dnlib will automatically remove the dead AssemblyRef.
